src/utils/contractFns/cancelTaskOnChain.ts

```diff
@@ -1,25 +1,36 @@
 import { AbiItem } from 'web3-utils';
-import { getWeb3Instance } from 'utils/web3EventFn';
+import { getArcanaWeb3Instance, getWeb3Instance } from 'utils/web3EventFn';
 import TaskAbi from 'contracts/abi/Task.sol/Task.json';
 import estimateGasPrice from 'utils/estimateGasFees';
+import { AuthContextType } from '@arcana/auth-react/types/typings';
 
 interface ICancelTaskOnChain {
   walletId: string | undefined;
   projectTaskAddress: string;
   selectedTask: any;
   builder: any;
+  auth: AuthContextType;
 }
 
 const cancelTaskOnChain = async ({
   walletId,
   projectTaskAddress,
   selectedTask,
   builder,
+  auth,
 }: ICancelTaskOnChain) => {
+  const arcanaWeb3Instance = await getArcanaWeb3Instance(auth);
+
   try {
     if (!walletId) throw new Error('Unable to complete the task');
 
-    const web3 = getWeb3Instance();
+    let web3;
+    if (arcanaWeb3Instance) {
```

Comment:
In what case this is a falsy value? Inside this value, it's doing `auth.connect()` and then returning the instance.